tests/test_base_db.py

```diff
@@ -0,0 +1,123 @@
+from sqlite3 import Connection
+
+import pytest
+import sqlean
+
+from overlore.sqlite.base_db import BaseDatabase
+from tests.utils.base_db_test_utils import (
+    given_insert_values,
+    given_update_values,
+    insert_data_into_db,
+    insert_query,
+    select_query,
+    update_query,
+)
+
+
+@pytest.fixture
+def db():
+    db = BaseDatabase.__new__(BaseDatabase)
+    db._init(":memory:", [], ["CREATE TABLE test_table (id INTEGER PRIMARY KEY, value TEXT)"], [], lambda db: None)
+    yield db
```

Comment:
The fixture is executed before every test right? So on the second and subsequent test, the function execution will start after the yield and just close the connection no?

Reply:
Execution flow is:
Setup phase:
yield -> execute test function
test function finishes
execute code after 'yield' statement

So whatever is done after the yield statement is used to as cleanup after the test has run

Reply:
Ah so the fixture executes before and after the test?

Reply:
Basically yeah,
anything before the yield is executed before the test, and anything after the yield is executed...after